### PR TITLE
Implementation of #5 and #6 features.

### PR DESCRIPTION
**Pull request by mmmm1998**

### CODE REVIEW

File: include/voxen/util/voxen_home.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <cassert>
+#include <filesystem>
+#include <voxen/config.hpp>
+
+using namespace std::filesystem;
```
Comment (Svenny) [inline include/voxen/util/voxen_home.hpp]: Avoid `using namespace` in public headers

File: src/common/config.cpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
+#include <voxen/common/config.hpp>
+
+#include <voxen/util/exception.hpp>
+#include <voxen/util/voxen_home.hpp>
+#include <voxen/util/log.hpp>
+
+#include <fmt/format.h>
+
+using namespace voxen;
```
Comment (Svenny) [inline src/common/config.cpp]: Replace `using namespace voxen` with wrapping the code in `namespace voxen {}`

File: src/common/config.cpp [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	Config::Scheme s;
+
+	s.push_back({"dev", "fps_logging", "This parameter enable fps and ups logging into log", false});
+	s.push_back({"window", "width", "This parameter control voxen window width", 1600L});
+	s.push_back({"window", "height", "This parameter control voxen window height", 900L});
+	s.push_back({"window", "fullscreen", "This parameter enable fullscreen for voxen window", false});
+	s.push_back({"controller", "mouse_sensitivity", "This parameter controls mouse sensitivity", 1.5});
+	s.push_back({"controller", "forward_speed", "This parameter controls player forward speed", 100.0});
+	s.push_back({"controller", "strafe_speed", "This parameter controls player strafe speed", 50.0});
+	s.push_back({"controller", "roll_speed", "This parameter controls player roll speed", 1.5 * 0.01});
```
Comment (Svenny) [inline src/common/config.cpp]: Replace to `s.emplace_back("section", "name", "comment", value);`

Comment (Svenny) [inline src/common/config.cpp]: "This parameter controls" prefix may be removed from comments - it's obvious and self-implied when reading config file

File: include/voxen/common/config.hpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+#include <variant>
+#include <string>
+#include <string_view>
+#include <vector>
+#include <map>
+
+#define SI_CONVERT_GENERIC
+#include <simpleini/SimpleIni.h>
+
+using std::string_view;
```
Comment (Svenny) [inline include/voxen/common/config.hpp]: Avoid `using` for importing types in public headers

File: include/voxen/common/config.hpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+#include <filesystem>
+#include <variant>
+#include <string>
+#include <string_view>
+#include <vector>
+#include <map>
+
+#define SI_CONVERT_GENERIC
+#include <simpleini/SimpleIni.h>
```
Comment (Svenny) [inline include/voxen/common/config.hpp:11]: Is it possible to move this include to `.cpp` file?

File: include/voxen/util/exception.hpp [hunk 6]
```diff
@@ -29,6 +30,19 @@ class MessageException : public Exception {
...
+class FormattedMessageException : public Exception {
+public:
+	explicit FormattedMessageException(std::string_view format_str, const fmt::format_args& format_args, const std::experimental::source_location &loc =
+	      std::experimental::source_location::current());
+	virtual ~FormattedMessageException() override = default;
+
+	virtual const char *what() const noexcept override;
+protected:
+	std::string m_what;
+	bool m_exception_accures;
```
Comment (Svenny) [inline include/voxen/util/exception.hpp]: A typo - change `accures` to `occures`

**Review by Svenny — CHANGES_REQUESTED** [review]
Generally OK, but I have some style-related propositions.

File: src/common/config.cpp (hunk 3 above)
Comment (Svenny) [reply src/common/config.cpp]: Well, actually this is not needed and requires a custom constructor for SchemeEntry.

File: include/voxen/common/config.hpp (hunk 5 above)
Comment (mmmm1998) [reply include/voxen/common/config.hpp:11]: Well, yes, but only via using 
````c++
std::unique_ptr<CSimpleIniA> m_ini;
````
or something similar

Comment (Svenny) [reply include/voxen/common/config.hpp:11]: Ok, we'll leave it as is for now.

**Review by Svenny — APPROVED** [review]

**Pull request merged**